File: 배선영/피보나치 수.java
class Solution {
    public int solution(int n) {
        int answer = 1;
        int pre = 1;
        int post = 1;
        
        for(int i= 3; i<n + 1; i++) {
            answer = (pre + post) % 1234567;
            pre = post;
            post = answer;
        }
        
        return answer;
    }
}

File: 신승혁/피보나치 수.js
const solution = (n) => {
  const dp = new Array(n + 1).fill(0);
  dp[1] = 1;

  for (let i = 2; i <= n; i++) {
    dp[i] = ((dp[i - 2] % 1234567) + (dp[i - 1] % 1234567)) % 1234567;
  }

  return dp[n];
};

const n = 5;
console.log(solution(5));
console.log(solution(100));
console.log(solution(1000));
console.log(solution(10000));
console.log(solution(100000));
